app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ProjectsController < ApplicationController
+  before_action :get_project, except: [:index, :create, :new]
+  before_action :authenticate_user!
+
+  def index
+    @projects = Project.all
+  end
+
+  def new
+    @project = Project.new
+  end
+
+  def create
+    @project = current_user.projects.create(project_params)
+    if @project.save
+      flash[:success] = t(:success_create)
+      redirect_to projects_path
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    if @project.update(project_params)
+      flash[:success] = t(:success_update)
+      redirect_to projects_path
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @project.destroy
+    flash[:success] = t(:success_destroy)
+    redirect_to projects_path
+  end
+
+  private
+
+  def project_params
+    params.require(:project).permit(:name, :description, :end_date, :start_date)
+  end
+
+  def get_project
+    @project = Project.find(params[:id])
```

Comment:
SECURITY ISSUE: If you write like this you can view or edit any project in system.